cmake/CMakeLists.txt

```diff
@@ -0,0 +1,112 @@
+set(LIBC_BOTTOM_HALF_HEADERS_PUBLIC_DIR ${PROJECT_SOURCE_DIR}/libc-bottom-half/headers/public)
+set(LIBC_TOP_HALF_MUSL_DIR ${PROJECT_SOURCE_DIR}/libc-top-half/musl)
+set(LIBC_TOP_HALF_MUSL_SRC_DIR ${LIBC_TOP_HALF_MUSL_DIR}/src)
+set(LIBC_BOTTOM_HALF_DIR ${PROJECT_SOURCE_DIR}/libc-bottom-half)
+set(LIBC_BOTTOM_HALF_CLOUDLIBC_SRC_DIR ${LIBC_BOTTOM_HALF_DIR}/cloudlibc/src)
+
+set(INSTALL_LIB_DIR lib/${MULTIARCH_TRIPLE})
+set(INSTALL_SHARE_DIR share/${MULTIARCH_TRIPLE})
+
+include(global-compiler-options.cmake)
+include(local-sysroot.cmake)
+include(predefined-macros.cmake)
+include(libc-imports.cmake)
+
+add_subdirectory(libc-bottom-half)
+add_subdirectory(libc-top-half)
+
+if(${MALLOC_IMPL} STREQUAL "dlmalloc")
+  add_subdirectory(dlmalloc)
+endif()
+
+add_library(wasi-emulated-signal)
+target_link_libraries(wasi-emulated-signal
+  wasi-emulated-signal-musl-objs
+  wasi-emulated-signal-objs
+)
+add_custom_command(TARGET wasi-emulated-signal POST_BUILD
+  COMMAND ${CMAKE_C_COMPILER_AR} crs $<TARGET_FILE:c>
+)
+
+add_library(c)
+
+set (LIBC_LINK_LIBRARIES
+  libc-bottom-half
+)
+
+if(${BUILD_LIBC_TOP_HALF} STREQUAL "ON")
+    list(APPEND LIBC_LINK_LIBRARIES
+      libc-top-half
+    )
+endif()
+
+if(${MALLOC_IMPL} STREQUAL "dlmalloc")
+  list(APPEND LIBC_LINK_LIBRARIES
+    dlmalloc)
+elseif(${MALLOC_IMPL} STREQUAL "none")
+  # nothing to do here
+else()
+  message(FATAL_ERROR "unknown malloc implementation ${MALLOC_IMPL}")
+endif()
+
+target_link_libraries(c
+  PRIVATE
+    ${LIBC_LINK_LIBRARIES}
+)
+
+add_custom_command(TARGET c POST_BUILD
+  COMMAND ${CMAKE_C_COMPILER_AR} crs $<TARGET_FILE:c>
+)
```

Comment:
Why do this?  Add a proper target for a static archive.  There is no guarantee the librarian accepts the GNU style arguments.